data/src/main/java/me/androidbox/data/alarm_manager/AlarmReceiver.kt

```diff
@@ -1,15 +1,75 @@
 package me.androidbox.data.alarm_manager
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import me.androidbox.data.R
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_AGENDA_TYPE
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_DESCRIPTION
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_ID
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_TITLE
 
 class AlarmReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
-        intent?.getStringExtra("EXTRA_MESSAGE")?.let { message ->
-            /* TODO Add Notification here */
-            println("Alarm triggered: $message")
+        /* TODO Not sure if intent and context can be null but I guess is ok to cast it anyway */
+        if(intent != null && context != null) {
+            intent.extras?.let { bundle ->
+                val agendaId = bundle.getString(EXTRA_ID) ?: ""
+                val title = bundle.getString(EXTRA_TITLE) ?: "Title"
+                val description = bundle.getString(EXTRA_DESCRIPTION) ?: "Description"
+                val agendaType = bundle.getString(EXTRA_AGENDA_TYPE) ?: "Agenda"
+
+                if (agendaId.isNotBlank()) {
+                    createNotificationChannel(context, agendaId, agendaType)
+                    createAndShowNotification(context, agendaId, title, description)
+                }
+            }
+        }
+    }
+
+    private fun createNotificationChannel(context: Context, agendaId: String, agendaName: String) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager = getNotificationManager(context)
+            val notificationChannel = NotificationChannel(
+                agendaId,
+                agendaName,
+                NotificationManager.IMPORTANCE_HIGH
+            )
+
+            notificationManager.createNotificationChannel(notificationChannel)
         }
     }
+
+    private fun createAndShowNotification(context: Context, agendaId: String, title: String, description: String) {
+        val notificationManager = getNotificationManager(context)
+        val notificationIntent = Intent(context, AlarmReceiver::class.java)
+        val pendingIntent = PendingIntent.getActivity(
+            context,
+            agendaId.hashCode(),
+            notificationIntent,
```

Comment:
Notificationintent is set to AlarmReceiver class

Won't clicking the notification do nothing? Or in fact, crash?

Reply:
@MKiperszmid @philipplackner  Should this be my `HomeActivity::class.java`

As I haven't completed the deeplinks yet. I just want the notification to do nothing if the user clicks it. Will that work

Reply:
This pending intent isn´t used for the time being, so can be removed and added later once you add deeplinks